types/tizen-tv-webapis/index.d.ts

```diff
@@ -1,61 +1,8744 @@
+    /**
+     * This method is callback parameter.
+     *
+     * @param data provide status
+     *
+     * @returns void
+     *
+     * @throw None N/A
+     *
+     * @version 3.0
+     *
+     * @since 2.4
+     */
+    (data: ShowRegisterPromotionalCodeData): void;
+}
+
+/**
+ * This callback interface defines credit card registration success callback.
+ *
+ * @since 2.4
+ */
+export interface BillingShowRegisterCreditCardCallback {
+    /**
+     * This method is callback parameter.
+     *
+     * @param data provide status
+     *
+     * @returns void
+     *
+     * @throw None N/A
+     *
+     * @version 3.0
+     *
+     * @since 2.4
+     */
+    (data: ShowRegisterCreditCardData): void;
+}
+
+/**
+ * This callback interface defines purchase history success callback.
+ *
+ * @since 2.4
+ */
+export interface BillingShowPurchaseHistoryCallback {
+    /**
+     * This method is callback parameter.
+     *
+     * @param data provide status
+     *
+     * @returns void
+     *
+     * @throw None N/A
+     *
+     * @version 3.0
+     *
+     * @since 2.4
+     */
+    (data: ShowPurchaseHistoryData): void;
+}
+
+/**
+ * Defines the service availability check success callback.
+ *
+ * @since 2.4
+ */
+export interface BillingIsServiceAvailableCallback {
+    /**
+     * Callback method returning the service availability check status.
+     *
+     * @param data It includes isServiceAvailable API result. but you can't use it as it is. you have to parse it as JSON format data.
+     *
+     * @returns void
+     *
+     * @throw None N/A
+     *
+     * @version 3.0
+     *
+     * @since 2.4
+     */
+    (data: ServiceAvailableData): void;
+}
+
+/**
+ * This module defines the Billing (Samsung Checkout) functionalities provided by the Tizen Samsung TV Product API.
+ *
+ * @privilegeLevel Public
+ *
+ * @privilegeName http://developer.samsung.com/privilege/billing
+ *
+ * @since 2.4
+ *
+ */
+export interface BillingManager {
+    /**
+     * Enables implementing the Samsung Checkout Client module within the application. After authenticating the purchase information through the application, the user can proceed to purchase payment.
+     *
+     * @param appId Application ID
+     *
+     * @param serverType Billing server type
+     *
+     * @param paymentDetails Payment parameters
+     * OrderItemID[MANDATORY]/OrderTitle[MANDATORY]/OrderTotal[MANDATORY]/OrderCurrencyID[MANDATORY]/OrderID[OPTIONAL]/OrderCustomID[MANDATORY]
+     *
+     * @param onsuccess Returns "payResult" and "payDetail" if there is no internal error occurs until client to server data communication.
+     * payResult, can still contains error when billing server confirms that the given parameters does not have expected value or have problem while processing it
+     * payDetail, can have additional data when it's returned, such as InvoiceID. Please refer to the development guide of "buyItem"	for details
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/tizen-tv-webapis/index.d.ts;
  Line: 2550;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.